src/lib.rs

```diff
@@ -58,10 +97,28 @@ fn relaunch() -> Result<Child, Error> {
         .map_err(Error::SpawnChild)
 }
 
+/// Graceful exit: We dispatch the singal that got to the application,
+/// followed by SIGTERM and SIGKILL.
 #[cfg(target_family = "unix")]
-fn pid1_handling(settings: &Pid1Settings, child: Option<Child>) -> ! {
+fn gracefull_exit(settings: Pid1Settings, signal: c_int, child_pid: i32) -> Result<(), Error> {
+    if signal == SIGINT {
+        let _ = kill(Pid::from_raw(child_pid), Some(nix::sys::signal::SIGINT));
+        std::thread::sleep(settings.timeout);
+    }
+    // Send SIGTERM to all the processes with pid > 1
+    let _ = kill(Pid::from_raw(child_pid), Some(nix::sys::signal::SIGTERM));
+    std::thread::sleep(settings.timeout);
+
+    // Okay, some children are still present. Use the SIGKILL card.
+    let _ = kill(Pid::from_raw(child_pid), Some(nix::sys::signal::SIGKILL));
+    std::thread::sleep(settings.timeout);
```

Comment:
Why sleep here? Shouldn't be necessary. If anything, I'd say sleep and then exit with an error code indicating that something impossible occurred.

Reply:
You're right, sleep is not required here. I have removed it, thanks!